core/src/main/java/com/datastax/oss/driver/api/core/cql/PreparedStatement.java

```diff
@@ -68,12 +69,40 @@
    */
   List<Integer> getPrimaryKeyIndices();
 
+  /**
+   * A unique identifier for result metadata (essentially a hash of {@link
+   * #getResultSetDefinitions()}).
+   *
+   * <p>This information is mostly for internal use: with protocol {@link CoreProtocolVersion#V5} or
+   * higher, the driver sends it with every execution of the prepared statement, to validate that
+   * its result metadata is still up-to-date.
+   *
+   * <p>Note: this method returns null for protocol {@link CoreProtocolVersion#V4} or lower;
+   * otherwise, the returned buffer is read-only.
+   *
+   * @see <a href="https://issues.apache.org/jira/browse/CASSANDRA-10786">CASSANDRA-10786</a>
+   */
+  ByteBuffer getResultMetadataId();
```

Comment:
Users can wrap prepared statements (and pass us a bound statement that references a wrapped prepared statement), so this has to be on the public API. Same for `setResultMetadata`.